### PR TITLE
Fix ssh completions and ssh provided hostnames

### PR DESCRIPTION
**Pull request by daleeidd**
This is a quick patch. Very sorry about missing this, but the ssh provided hostnames did not include the root ssh config – only the hostnames discovered through files included with the `Include` directive.

Also, if an `Include` directive contained a wildcard which resolved to nothing, `echo` would complain which this patch fixes as well.

### CODE REVIEW

File: share/functions/__fish_print_hostnames.fish [hunk 1]
```diff
@@ -59,17 +59,25 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
...
-                set paths (eval "echo $paths" | string split ' ')
-                for path_index in (seq (count $paths))
-                    # Resolve relative paths
-                    if string match -v '/*' $paths[$path_index] >/dev/null
-                        set paths[$path_index] $relative_path/$paths[$path_index]
+
+            set -l new_paths
+            for path in $paths
+                # Guard against echo command complaining on zero wildcard matches
+                if test (eval "count $path") -gt 0
```
Comment (krader1961) [inline share/functions/__fish_print_hostnames.fish]: Use `set` to inhibit the glob match failure:

```
set -l expanded_path
eval set expanded_path (echo $path)
for path in $expanded_path
```

This is a good example for why we should implement the proposed `glob` command.

Comment (daleeidd) [inline share/functions/__fish_print_hostnames.fish]: Perfect. Thank you. I have amended the commit.

File: share/functions/__fish_print_hostnames.fish [hunk 2]
```diff
@@ -59,17 +59,23 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
...
-                    if string match -v '/*' $paths[$path_index] >/dev/null
-                        set paths[$path_index] $relative_path/$paths[$path_index]
+
+            set -l new_paths
+            for path in $paths
+                set -l expanded_path
+                eval set expanded_path (echo $path)
+                for path in $expanded_path
+                    # Resolve "relative" paths in accordance to ssh path resolution
+                    if string match -v '/*' $path >/dev/null
```
Comment (krader1961) [inline share/functions/__fish_print_hostnames.fish]: Rather than redirect the output use the `-q` flag.

**Comment by krader1961** [comment]
Other than one minor nit about using `-q` rather than redirecting the output of `string match` this LGTM. @faho, or anyone else, see any problems with this?

File: share/functions/__fish_print_hostnames.fish (hunk 2 above)
Comment (daleeidd) [inline share/functions/__fish_print_hostnames.fish]: Thank you. Fixed.

**Comment by faho** [comment]
@krader1961: That was the only thing I noticed, so LGTM. And yes, that `eval` is needed here is quite annoying.

**Comment by faho** [comment]
Merged, thanks!

I'm working on a `expand` builtin that could be used instead of `eval` here.

**Pull request merged**